<h1>Segmenting and Clustering Neighborhoods in the city of Toronto, Canada</h1>

<h3> Coursera - IBM Applied Data Science Capstone - Week 3 PART 1</h3>

In this task, Toronto neighbourhoods are explored, segmented and clustered  based on the postalcode and borough information.

<h4><strong>1. Import libraries</strong></h4>

In [1]:
!pip install beautifulsoup4
!pip install lxml
import requests
import pandas as pd
from bs4 import BeautifulSoup

     |████████████████████████████████| 122kB 11.8MB/s eta 0:00:01


<h4><strong>2. Data Scraping</strong></h4>

For the Toronto neighborhood data, a Wikipedia page exists that has all the information needed to explore and cluster the neighborhoods in Toronto.


In [2]:
url='https://en.wikipedia.org/w/index.php?title=List_of_postal_codes_of_Canada:_M&oldid=890001695'

Next steps are to scrape the Wikipedia page and wrangle the data, clean it, and then read it into a pandas dataframe so that it is in a structured format.

In [3]:
response = requests.get(url)

data = response.text
soup = BeautifulSoup(data,'html.parser')
tor_table=soup.find('table')

df = pd.read_html(str(tor_table))[0]
df.head()

Postcode           Borough     Neighbourhood
0      M1A      Not assigned      Not assigned
1      M2A      Not assigned      Not assigned
2      M3A        North York         Parkwoods
3      M4A        North York  Victoria Village
4      M5A  Downtown Toronto      Harbourfront

The dataframe will consist of three columns: PostCode, Borough, and Neighborhood

<h4><strong>3. Cleaning Data</strong></h4>

Only process the cells that have an assigned borough. Ignore cells with a borough that is Not assigned.

In [4]:
df_cleaned=df[df['Borough'].str.contains("Not assigned") == False].reset_index()
df_cleaned.head()

index Postcode           Borough     Neighbourhood
0      2      M3A        North York         Parkwoods
1      3      M4A        North York  Victoria Village
2      4      M5A  Downtown Toronto      Harbourfront
3      5      M5A  Downtown Toronto       Regent Park
4      6      M6A        North York  Lawrence Heights

More than one neighbourhood can exist in one postal code area. These two rows will be combined into one row</br> with the neighbourhoods separated with a comma.

In [5]:
df_toronto= df_cleaned.groupby(['Postcode', 'Borough'])['Neighbourhood'].apply(', '.join).reset_index()
df_toronto.head()

Postcode      Borough                           Neighbourhood
0      M1B  Scarborough                          Rouge, Malvern
1      M1C  Scarborough  Highland Creek, Rouge Hill, Port Union
2      M1E  Scarborough       Guildwood, Morningside, West Hill
3      M1G  Scarborough                                  Woburn
4      M1H  Scarborough                               Cedarbrae

In [6]:
# Use the .shape method to print the number of rows of your dataframe
df_toronto.shape

(103, 3)

The table has 103 rows.

<h4><strong>4. Save Data into CSV File</strong></h4>

In [7]:
df_toronto.to_csv('Capstone_week3_part1.csv')